### PR TITLE
Fix sign up UI

### PR DESCRIPTION
**Pull request by swashcap**
* **Problem:** Signing up via the UI doesn't work
* **Solution:**
  * Ensure _coinstac-client-core_ authenticates newly created users
  * Ensure user is placed in app’s render process’s state upon signup
* **Testing:**
  1. Build a client from this branch:
    ```shell
    git fetch --all
    git checkout bug/#82-account-creation
    cd packages/coinstac-ui
    npm run build
    ```
  2. Clean local state: `npm run clean:db`
  3. Boot the client and attempt to make a new account

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/173590139517418/211870140170035)

### CODE REVIEW

File: packages/coinstac-client-core/src/index.js [hunk 1]
```diff
@@ -225,10 +225,11 @@ class CoinstacClient {
...
    */
   _initAuthorization(credentials) {
-    if (credentials.email && credentials.name) {
-      return this.auth.createUser(credentials);
-    }
-    return this.auth.login(credentials);
+    const doLogin = () => this.auth.login(credentials);
+
+    return credentials.email && credentials.name ?
+      this.auth.createUser(credentials).then(doLogin) :
```
Comment (swashcap) [inline packages/coinstac-client-core/src/index.js:231]: Big change in _coinstac-client-core_ is here: create that user, then do that login.

File: packages/coinstac-client-core/src/sub-api/auth.js [hunk 2]
```diff
@@ -70,34 +64,27 @@ class Auth {
...
-      if (err.status === 500) {
-        throw new Error(
-          'unable to log into COINS.  Please try again or file a support ticket'
-        );
-      } else if (err.status === 401) {
-        throw new Error('Invalid username or password.  Please try again');
-      }
-      throw err;
-    };
+  login({ password, username }) {
```
Comment (swashcap) [inline packages/coinstac-client-core/src/sub-api/auth.js:74]: Cleanup of general grossness.

File: packages/coinstac-client-core/test/coinstac-client.js [hunk 3]
```diff
@@ -213,3 +213,33 @@ test('CoinstacClient - teardown process', (t) => {
     .catch(t.end);
 });
 
+test('CoinstacClient - auth initialization', t => {
```
Comment (swashcap) [inline packages/coinstac-client-core/test/coinstac-client.js:216]: Ensures login is called when creating a user. Not the greatest test, I think, but it works.

File: packages/coinstac-ui/app/main/utils/boot/parse-cli-input.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ function parse() {
         data = data.toString();
         // don't fill the screen with useful build information. scan for it,
         // summarize it instead
-        const builtRegex = /.*(\[(built|not cacheable)\]|\{0\}).*[\r\n]/gm;
+        const builtRegex = /.*(\[(built|not cacheable)]|\{0\}).*[\r\n]/gm;
```
Comment (swashcap) [inline packages/coinstac-ui/app/main/utils/boot/parse-cli-input.js:30]: I guess that `\` isn’t necessary when the first `[` is escaped? Linter barked at it.

File: packages/coinstac-ui/app/render/components/form-signup-controller.js [hunk 5]
```diff
@@ -43,12 +43,23 @@ class FormSignupController extends Component {
...
-    }));
+
+    if (error) {
+      return FormSignupController.handleSignupError(error);
+    }
+
+    return dispatch(signUp(formData))
+      .then(() => {
+        app.notifications.push({
+          autoDismiss: 2,
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/form-signup-controller.js:54]: Notification goes away after 2 seconds. 👋😬 We should make this the default.

File: packages/coinstac-ui/app/render/state/ducks/auth.js [hunk 6]
```diff
@@ -24,13 +24,13 @@ export const logout = applyAsyncLoading(() => {
   };
 });
 
-export const signUp = applyAsyncLoading(user => {
-  return (dispatch) => { // eslint-disable-line
-    return app.core.initialize(user)
-    .then(() => app.notify('success', 'New user created successfully'))
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/state/ducks/auth.js:30]: Moved notification calls to the component. This makes things clearer: folks look at the component to see the control flow, everything in _app/render/state/_ should deal with **state only**.

File: packages/coinstac-ui/app/render/state/ducks/util.js [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import app from 'ampersand-app';
 
 export const notifyAndThrow = (err) => {
+  app.logger.error(err);
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/state/ducks/util.js:4]: Make this slightly less useless. Log it.

File: packages/coinstac-ui/app/render/components/form-signup-controller.js (hunk 5 above)
Comment (rssk) [reply packages/coinstac-ui/app/render/components/form-signup-controller.js:54]: Gah, totally

File: packages/coinstac-client-core/src/sub-api/auth.js [hunk 8]
```diff
@@ -35,27 +33,23 @@ class Auth {
...
-    // @TODO use actual user siteId
-    /* istanbul ignore next */
-    return bluebird.resolve(this.halfpenny.users.post(newUser));
+  createUser({ email, name, password, username }) {
+    return this.halfpenny.users.post({
+      email,
+      label: name,  // the API requires a `label` key instead of `name`
+      password,
+      siteId: '7',
+      username,
```
Comment (rssk) [inline packages/coinstac-client-core/src/sub-api/auth.js]: username and pass need to be base64 :-) https://github.com/MRN-Code/quarterback/blob/master/packages/steelpenny/src/lib/controllers/users.js#L144

This is newish for the PR site

Comment (swashcap) [reply packages/coinstac-client-core/src/sub-api/auth.js]: Fixed in d50e0c74a4e0835b5ac9ecbb978f26a891536384.

**Pull request merged**